lib/mercator.js

```diff
@@ -0,0 +1,283 @@
+/**
+ * @typedef {Object} Point
+ * @property {Number} x
+ * @property {Number} y
+ */
+
+/**
+ * @typedef {Object} LatLng
+ * @property {Number} lat
+ * @property {Number} lng
+ */
+
+/**
+ * @typedef {Object} Tile
+ * @property {Number} x
+ * @property {Number} y
+ * @property {Number} z
+ */
+
+/**
+ * @typedef {Object} TileBounds
+ * @property {LatLng} sw
+ * @property {LatLng} ne
+ */
+
+/**
+ * @typedef {Object} Pixel
+ * @property {Number} x
+ * @property {Number} y
+ */
+
+/**
+ * @typedef {Object} Evt
+ * @property {Pixel} pixel
+ * @property {LatLng} latLng
+ */
+
+/**
+ * Convert a `LatLng` Google Maps object to a `Point` object.
+ * @param {google.maps.LatLng} latLng
+ * @return {Point}
+ */
+const fromLatLngToPoint = (latLng) => {
+  const siny = Math.min(Math.max(Math.sin(latLng.lat() * (Math.PI / 180)), -.9999), .9999);
+
+  return {
+    x: 128 + latLng.lng() * (256 / 360),
+    y: 128 + 0.5 * Math.log((1 + siny) / (1 - siny)) * -(256 / (2 * Math.PI)),
```

Comment:
I'm suspicious of those magic 128 and 256 numbers. I feel like it's probably hard-coded for a 256 tile size, even though other parts of the library allow you to set the tile size. 

Nothing to change now, but certainly something to be wary of.